src/util/text-utils-test.cc

```diff
@@ -173,7 +173,81 @@ void TestConvertStringToReal() {
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE inf", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGEinf", &d));
+  KALDI_ASSERT(!ConvertStringToReal("infGARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("inf_GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("inf GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE infinity", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGEinfinity", &d));
+  KALDI_ASSERT(!ConvertStringToReal("infinityGARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("infinity_GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("infinity GARBAGE", &d));
+  KALDI_ASSERT(ConvertStringToReal("1.#INF", &d) && d > 0 && d - d != 0);
+  KALDI_ASSERT(ConvertStringToReal("-1.#INF", &d) && d < 0 && d - d != 0);
+  KALDI_ASSERT(ConvertStringToReal("-1.#INF  ", &d) && d < 0 && d - d != 0);
+  KALDI_ASSERT(ConvertStringToReal(" -1.#INF ", &d) && d < 0 && d - d != 0);
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE 1.#INF", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE1.#INF", &d));
+  KALDI_ASSERT(!ConvertStringToReal("2.#INF", &d));
+  KALDI_ASSERT(!ConvertStringToReal("-2.#INF", &d));
+  KALDI_ASSERT(!ConvertStringToReal("1.#INFGARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("1.#INF_GARBAGE", &d));
+
   KALDI_ASSERT(ConvertStringToReal("nan", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("+nan", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("-nan", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("Nan", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("NAN", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("NaN", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal(" NaN", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("NaN ", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal(" NaN ", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("1.#QNAN", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("-1.#QNAN", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal("1.#QNAN  ", &d) && d != d);
+  KALDI_ASSERT(ConvertStringToReal(" 1.#QNAN ", &d) && d != d);
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE nan", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGEnan", &d));
+  KALDI_ASSERT(!ConvertStringToReal("nanGARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("nan_GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("nan GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE 1.#QNAN", &d));
+  KALDI_ASSERT(!ConvertStringToReal("GARBAGE1.#QNAN", &d));
+  KALDI_ASSERT(!ConvertStringToReal("2.#QNAN", &d));
+  KALDI_ASSERT(!ConvertStringToReal("-2.#QNAN", &d));
+  KALDI_ASSERT(!ConvertStringToReal("-1.#QNAN_GARBAGE", &d));
+  KALDI_ASSERT(!ConvertStringToReal("-1.#QNANGARBAGE", &d));
+}
+
+template<class Real>
+void TestNan() {
+  Real d;
+  std::ostringstream strs;
+  strs << sqrt(-1);
+  std::string s = strs.str();
+
+  KALDI_ASSERT(ConvertStringToReal(s, &d) && d != d);
+}
+
+template<class Real>
+void TestInf() {
+  Real d;
+  std::ostringstream strs;
+  strs << 1.0/0.0;
```

Comment:
I don't think this is a safe way to get infinity.
According to wikipedia:
`Depending on the programming environment and the type of number (e.g. floating point, integer) being divided by zero, it may generate positive or negative infinity by the IEEE 754 floating point standard`.
Try exp(10000).